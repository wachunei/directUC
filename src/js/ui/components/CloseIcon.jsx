import React from "react";

export default function CloseIcon({ ...props }) {
  return (
    <svg
      width="19"
      height="19"
      viewBox="0 0 19 19"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M17 2L9.5 9.5m0 0L2 17m7.5-7.5L2 2m7.5 7.5L17 17"
        stroke="#828282"
        strokeWidth="3.089"
      />
    </svg>
  );
}
