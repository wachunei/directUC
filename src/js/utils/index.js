export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
