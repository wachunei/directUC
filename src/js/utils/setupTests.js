// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
