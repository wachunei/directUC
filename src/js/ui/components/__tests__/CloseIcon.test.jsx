import React from "react";
import CloseIcon from "../CloseIcon";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("CloseIcon Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<CloseIcon />);
    expect(tree).toMatchSnapshot();
  });

  test("should accept props", () => {
    const tree = renderTree(<CloseIcon width="200px" height="200px" />);
    expect(tree).toMatchSnapshot();
  });
});
