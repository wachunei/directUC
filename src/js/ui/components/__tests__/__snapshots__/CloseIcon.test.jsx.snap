// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon Component should accept props 1`] = `
<svg
  fill="none"
  height="200px"
  viewBox="0 0 19 19"
  width="200px"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M17 2L9.5 9.5m0 0L2 17m7.5-7.5L2 2m7.5 7.5L17 17"
    stroke="#828282"
    stroke-width="3.089"
  />
</svg>
`;

exports[`CloseIcon Component should match snapshot 1`] = `
<svg
  fill="none"
  height="19"
  viewBox="0 0 19 19"
  width="19"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M17 2L9.5 9.5m0 0L2 17m7.5-7.5L2 2m7.5 7.5L17 17"
    stroke="#828282"
    stroke-width="3.089"
  />
</svg>
`;

exports[`CloseIcon Component should match snapshot with darkTheme 1`] = `
<svg
  fill="none"
  height="19"
  viewBox="0 0 19 19"
  width="19"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M17 2L9.5 9.5m0 0L2 17m7.5-7.5L2 2m7.5 7.5L17 17"
    stroke="#828282"
    stroke-width="3.089"
  />
</svg>
`;

exports[`CloseIcon Component should match snapshot with lightTheme 1`] = `
<svg
  fill="none"
  height="19"
  viewBox="0 0 19 19"
  width="19"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M17 2L9.5 9.5m0 0L2 17m7.5-7.5L2 2m7.5 7.5L17 17"
    stroke="#828282"
    stroke-width="3.089"
  />
</svg>
`;
